### PR TITLE
Added quiz about ssh and permissions

### PR DESCRIPTION
**Pull request by cfculler**
Tests knowledge of Unix file permissions and basic reasoning behind ssh key exchanges

### CODE REVIEW

File: src/quizzes/chapter-1/1.3.4/sshconfig-quiz.js [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+	> Using key exhanges are *more* secure than passwords, because they are longer, more complex, and encrypted
+- [ ] They allow multiple users to share the same authentication credentials, which enhances security
+	> This is actually harmful for security. If one user's information leaks, it compromises others aswell
+
+# You read above that the \`.ssh\` directory should have permissions set to 0700 and the \`.ssh/config\` file should have permissions set to 0600. Why should the one's and ten's place be set to 0?<br>
+You may choose to explore more about permissions <a href="https://mason.gmu.edu/~montecin/UNIXpermiss.htm">here</a>
+
+1. [x] The hundred's place resembles owner permissions, and groups and others shouldn't have access to these, so they are set to 0
+1. [ ] The hundred's place resembles group permissions, and owners and others shouldn't have access to these, so they are set to 0
+1. [ ] The hundred's place resembles other's (not owner or group) permissions, and owners and groups shouldn't have access to these, so they are set to 0
```
Comment (jburns24) [inline src/quizzes/chapter-1/1.3.4/sshconfig-quiz.js:29]: Let's replace this question with:

```
Why are the permission levels `0700` for the `.ssh/` directory and `0600` for the `.ssh/authorized_keys` file recommended as secure settings for SSH authentication? Select the most accurate explanation.

1. [ ] To ensure that `.ssh/` and `.ssh/authorized_keys` are readable, writable, and executable by everyone, guaranteeing maximum compatibility across different users and systems.
2. [x] To allow only the file owner to read and write `.ssh/authorized_keys` and to access the `.ssh/` directory, preventing unauthorized users from viewing or modifying private SSH keys.
3. [ ] To enable group and other users to access SSH keys for administrative and troubleshooting purposes, while restricting write permission to the owner only.
4. [ ] To restrict all access to `.ssh/` and `.ssh/authorized_keys` for security purposes, making these files and directories completely inaccessible.
```

File: src/quizzes/chapter-1/1.3.4/sshconfig-quiz.js [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+# What does the 0700 permission denote for the \`.ssh\` directory?<br>
+You may choose to explore more about permissions <a href="https://mason.gmu.edu/~montecin/UNIXpermiss.htm">here</a>
+
+1. [x] Read, Write, and Execute permissions
+1. [ ] Read and Write permissions
+1. [ ] Read and Execute permissions
+1. [ ] Write and Execute permissions
+1. [ ] Read permissions
+1. [ ] Write permissions
+1. [ ] Execute permissions
```
Comment (jburns24) [inline src/quizzes/chapter-1/1.3.4/sshconfig-quiz.js:40]: Let's change this up a bit so the selections probe if they know _who_ is getting those permissions (ie: owner group or public)

Maybe the options are 
```
[x] Owner has Read, Write, and Execute permissions
[] Group has Read, Write, and Execute permissions
[] Owner has Read  and Execute permissions
[] Group has Read, and Execute permissions
```

File: src/quizzes/chapter-1/1.3.4/sshconfig-quiz.js [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+# What does the 0600 permission denote for the \`.ssh/config\` file?
+1. [ ] Read, Write, and Execute permissions
+1. [x] Read and Write permissions
+1. [ ] Read and Execute permissions
+1. [ ] Write and Execute permissions
+1. [ ] Read permissions
+1. [ ] Write permissions
+1. [ ] Execute permissions
```
Comment (jburns24) [inline src/quizzes/chapter-1/1.3.4/sshconfig-quiz.js:49]: Same comment as above. Let's change the options to probe if they know who has what permissions

Comment (jburns24) [reply src/quizzes/chapter-1/1.3.4/sshconfig-quiz.js:49]:
```
```
[] Owner has Read, Write, and Execute permissions
[] Group has Read, Write, and Execute permissions
[] Owner has Read  and Execute permissions
[] Group has Read, and Execute permissions
[x] Group has Read and Write permissions
[] Owner has Read Write permissions
```

File: src/quizzes/chapter-1/1.3.4/sshconfig-quiz.js (hunk 1 above)
Comment (cfculler) [reply src/quizzes/chapter-1/1.3.4/sshconfig-quiz.js]: Made this change. I like it

File: src/quizzes/chapter-1/1.3.4/sshconfig-quiz.js (hunk 2 above)
Comment (cfculler) [reply src/quizzes/chapter-1/1.3.4/sshconfig-quiz.js]: Made this change too. Still gets my initial point across with the question but checks additional understanding.

File: src/quizzes/chapter-1/1.3.4/sshconfig-quiz.js (hunk 3 above)
Comment (cfculler) [reply src/quizzes/chapter-1/1.3.4/sshconfig-quiz.js]: I made this change as well.

**Review by jburns24 — APPROVED** [review]

**Pull request merged**